File: TyTraCLTransformations/MemoryReduction/test_return_to_array_elt.c
#include <stdio.h>

void square(int,int*);

int main() {
    int v[16];
    int i, n;
    for (i=0;i<16;++i) {
        v[i]=0;
    }
    n=1;
    for ( i=0;i<4; ++i) {
        square(n,&v[n-1]);
        square(n,&v[i]);
        n=n*2;
    }
    for (i=0;i<16;++i) {
        printf("%d\n",v[i]);
    }
    return 0;
}

void square(int x_in,int* x_out) {
    *x_out = x_in * x_in;
}    

File: TyTraCLTransformations/MemoryReduction/test_return_to_array_elt.f90
! gfortran test_return_to_array_elt.f90
program testReturnToArrayElt
    integer, dimension(16) :: v
    integer i, n
    v=0
    n=1
    do i=1,4    
        call square(n,v(n))
        call square(n,v(i))
        n=n*2
    end do
    print *,v
end program testReturnToArrayElt

subroutine square(x_in,x_out)
    integer, intent(in) :: x_in
    integer, intent(out) :: x_out
    x_out = x_in*x_in
end subroutine square    
    
